paddle/inference/inference.cc

```diff
@@ -25,19 +25,37 @@ limitations under the License. */
 
 namespace paddle {
 
+void InferenceEngine::LoadInferenceModel(const std::string& dirname) {
+  std::string model_filename = dirname + "/__model__.dat";
+  LOG(INFO) << "loading model from " << model_filename;
+  std::ifstream inputfs(model_filename, std::ios::in | std::ios::binary);
+  std::string program_desc_str;
+  inputfs.seekg(0, std::ios::end);
+  program_desc_str.resize(inputfs.tellg());
+  inputfs.seekg(0, std::ios::beg);
+  LOG(INFO) << "program_desc_str's size: " << program_desc_str.size();
+  inputfs.read(&program_desc_str[0], program_desc_str.size());
+  inputfs.close();
+
+  program_ = new framework::ProgramDesc(program_desc_str);
+  GenerateLoadProgram(dirname);
+
+  framework::BlockDesc* global_block = program_->MutableBlock(0);
+  feed_var_names_.clear();
+  fetch_var_names_.clear();
+  for (auto* op : global_block->AllOps()) {
+    if (op->Type() == "feed") {
+      feed_var_names_.insert(feed_var_names_.begin(), op->Output("Out")[0]);
+    } else if (op->Type() == "fetch") {
+      fetch_var_names_.push_back(op->Input("X")[0]);
+    }
+  }
+}
+
 void InferenceEngine::LoadInferenceModel(
```

Comment:
I think we can remove this function. If there are not `feed_op` and `fetch_op` in the `ProgramDesc`, users can specify these when calling `Run()`.

Reply:
Sorry, i don't understand this properly. Based on the updated design, the `Run()` function does not take as input the vector of fetch_var_name and feed_var_names. Right?
```  
void Run(const ProgramDesc* program,
           Scope* scope,
           std::map<std::string, Tensor>& feeds,
           std::map<std::string, Tensor>& fetchs,
           std::string& feed_var_name = "feed",
           std::string& fetch_var_name = "fetch") {
```

Reply:
We can get the `feed_var_names` from the argument `std::map<std::string, Tensor>& feeds`, where the `std::string` represent a `feed_var_name` and the `Tensor` is input data.

Reply:
okay, will take a look. Thanks for the reply.

Reply:
Yes, will remove this function in the next PR.